source/hwPortUtils.py

```diff
@@ -438,22 +439,20 @@ def _getHidInfo(hwId, path):
 	if hwId.startswith("VID"):
 		info["provider"] = "usb"
 		info["usbID"] = hwId[:17] # VID_xxxx&PID_xxxx
-		return info
-	if not hwId.startswith("{00001124-0000-1000-8000-00805f9b34fb}"): # Not Bluetooth
+	elif hwId.startswith("{00001124-0000-1000-8000-00805f9b34fb}"):  # Not Bluetooth
+		info["provider"] = "bluetooth"
+	else:
 		# Unknown provider.
 		info["provider"] = None
 		return info
-	info["provider"] = "bluetooth"
 	# Fetch additional info about the HID device.
-	# This is a bit slow, so we only do it for Bluetooth,
-	# as this info might be necessary to identify Bluetooth devices.
 	from serial.win32 import CreateFile, INVALID_HANDLE_VALUE, FILE_FLAG_OVERLAPPED
 	handle = CreateFile(path, 0,
 		winKernel.FILE_SHARE_READ | winKernel.FILE_SHARE_WRITE, None,
 		winKernel.OPEN_EXISTING, FILE_FLAG_OVERLAPPED, None)
 	if handle == INVALID_HANDLE_VALUE:
 		if _isDebug():
-			log.debug(u"Opening device {dev} to get additional info failed: {exc}".format(
+			log.debugWarning(u"Opening device {dev} to get additional info failed: {exc}".format(

```

Comment:
```suggestion
			log.debugWarning("Opening device {dev} to get additional info failed: {exc}".format(
```
Or alternatively, use an fstring instead of .format